app/controllers/providers/gateway_evidences_controller.rb

```diff
@@ -0,0 +1,107 @@
+module Providers
+  class GatewayEvidencesController < ProviderBaseController
+    def show
+      populate_form
+    end
+
+    def update
+      if upload_button_pressed?
+        perform_upload
+      elsif save_or_continue
+        convert_new_files_to_pdf
+      else
+        render :show
+      end
+    end
+
+    def destroy
+      original_file = delete_original_and_pdf_files
+      @successfully_deleted = files_deleted_message(original_file.attachment_name) unless original_file.nil?
+      populate_form
+      render :show
+    end
+
+    private
+
+    def save_or_continue
+      if form.files?
+        save_continue_or_draft(form)
+      else
+        legal_aid_application.reload
```

Comment:
Is this a good solution to stop the application saving an obsolete gateway evidence record if there are no files attached?

Reply:
I don't think you need to reload it here.  From what I can see, the `GatewayEvidence` record is only created when you've pressed the upload button for the first time. So I think, when you're here and `form.files?` returns false, then the `GatewayEvidence` record hasn't been built and attached to the `LegalAidApplication` and you could just return nil.  Or am I missing something?